src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -246,39 +246,39 @@ private void tick() {
              * modifies the collection during iteration.
              */
             queueLock.writeLock().lock();
-
+            Map<Location, Boolean> deletionQueueCopy;
             try {
-                Iterator<Map.Entry<Location, Boolean>> removals = deletionQueue.entrySet().iterator();
-
-                while (removals.hasNext()) {
-                    Map.Entry<Location, Boolean> entry = removals.next();
-                    BlockStorage.deleteLocationInfoUnsafely(entry.getKey(), entry.getValue());
-                    removals.remove();
-                }
+                // Copy the queue, this way we don't have keep the queue locked whilst removing
+                deletionQueueCopy = new HashMap<>(deletionQueue);
+                deletionQueue.clear();
```

Comment:
This is a good change, we shouldn't keep this locked for the (heavy as fuck) BlockStorage modifications.